app/presenters/content_item/contents_list.rb

```diff
@@ -3,24 +3,58 @@ module ContentsList
     include ActionView::Helpers::UrlHelper
     include TypographyHelper
 
+    CHARACTER_LIMIT = 100
+
     def contents
-      @contents ||= contents_items.each do |item|
-        item[:href] = "##{item[:id]}"
-      end
+      @contents ||=
+        if show_contents_list?
+          contents_items.each { |item| item[:href] = "##{item[:id]}" }
+        else
+          []
+        end
     end
 
     def contents_items
-      extract_headings_with_ids(body)
+      extract_headings_with_ids
+    end
+
+    def show_contents_list?
+      return false if contents_items.count < 2
+      return true if contents_items.count > 2
```

Comment:
what if contents_items.count == 2?

Reply:
Line 24 would catch this case but only if it's a paragraph - @emmabeynon is it ever likely to be a 2 item list which doesn't contain paragraphs?

Reply:
Subsequent commits deal with checking for content with a lot of table rows, and also content with a mix of images, long content and long tables.

Reply:
Makes sense, I hadn't looked ahead enough, sorry.